Source/WebCore/platform/graphics/cg/PDFDocumentImage.cpp

```diff
@@ -97,94 +80,6 @@ EncodedDataStatus PDFDocumentImage::dataChanged(bool allDataReceived)
-{
-    // Old and new source rectangles have to match.
-    if (srcRect != m_cachedSourceRect)
-        return false;
-
-    // Old and new scaling factors "dest / src" have to match.
-    if (dstRect.size() != m_cachedDestinationRect.size())
-        return false;
-
-    // m_cachedImageRect can be moved if srcRect and dstRect.size() did not change.
-    FloatRect movedCachedImageRect = m_cachedImageRect;
-    movedCachedImageRect.move(FloatSize(dstRect.location() - m_cachedDestinationRect.location()));
-
-    // movedCachedImageRect has to contain the whole dirty rectangle.
-    FloatRect dirtyRect = intersection(context.clipBounds(), dstRect);
-    if (!movedCachedImageRect.contains(dirtyRect))
-        return false;
-
-    // Old and new context scaling factors have to match as well.
-    AffineTransform::DecomposedType decomposedTransform;
-    context.getCTM(GraphicsContext::DefinitelyIncludeDeviceScale).decompose(decomposedTransform);
-
-    AffineTransform::DecomposedType cachedDecomposedTransform;
-    m_cachedTransform.decompose(cachedDecomposedTransform);
-    if (decomposedTransform.scaleX != cachedDecomposedTransform.scaleX || decomposedTransform.scaleY != cachedDecomposedTransform.scaleY)
-        return false;
-
-    return true;
-}
-
-static void transformContextForPainting(GraphicsContext& context, const FloatRect& dstRect, const FloatRect& srcRect)
-{
-    float hScale = dstRect.width() / srcRect.width();
-    float vScale = dstRect.height() / srcRect.height();
-
-    if (hScale != vScale) {
-        float minimumScale = std::max((dstRect.width() - 0.5) / srcRect.width(), (dstRect.height() - 0.5) / srcRect.height());
-        float maximumScale = std::min((dstRect.width() + 0.5) / srcRect.width(), (dstRect.height() + 0.5) / srcRect.height());
-
-        // If the difference between the two scales is due to integer rounding of image sizes,
-        // use the smaller of the two original scales to ensure that the image fits inside the
-        // space originally allocated for it.
-        if (minimumScale <= maximumScale) {
-            hScale = std::min(hScale, vScale);
-            vScale = hScale;
-        }
-    }
-
-    // drawPDFPage() relies on drawing the whole PDF into a context starting at (0, 0). We need
-    // to transform the destination context such that srcRect of the source context will be drawn
-    // in dstRect of destination context.
-    context.translate(dstRect.location() - srcRect.location());
-    context.scale(FloatSize(hScale, -vScale));
-    context.translate(0, -srcRect.height());
-
-    // FIXME https://bugs.webkit.org/show_bug.cgi?id=227808: Surely there needs to be a clip operation here,
-    // if the srcRect is smaller than the whole PDF.
-}
-
-// To avoid the jetsam on iOS, we are going to limit the size of all the PDF cachedImages to be 64MB.
-static const size_t s_maxCachedImageSide = 4 * 1024;
```

Comment:
Do we just not need this code any more? Are we going to trigger more iOS jetsams by removing it?

Reply:
I do not think so. The maximum area of ImageViewport in device pixels is 
```
max_viewport_area = ImageBuffer::maxClampedSide x ImageBuffer::maxClampedSide
```
which is the same as the old code see `calculateViewportSourceRect()`.

Also this maximum area 64MB is the maximum of the 2D canvas on iOS, see `maxCanvasArea()`. So if we are allowing this size for a 2D canvas on iOS, the PDF image will have the same memory cost. But this is in addition to the encoded data of the PDF file and the native image we create for the image which is drawn on the tile.

Reply:
I made the maximum size of the ImageBuffer stored by CachedSubimage in bytes:

```
    // The tile size is usually 512 x 512, account for the 2x display resolution
    // and make room for one more tile in every direction.
    static constexpr float maxSide = 512 * 2 * 3
    static constexpr float maxArea = maxSide * maxSide;
    static constexpr float maxMemory = maxArea * 4;
```

This is 36MB. And this gives a higher cache usage which is a little bit more than 2. (create 100 sublimates and draw 200 times).